pkg/xds/generator/outbound_proxy_generator.go

```diff
@@ -213,11 +226,20 @@ func (_ OutboundProxyGenerator) determineSubsets(proxy *model.Proxy, outbound *k
 			// 0 assumes no traffic is passed there. Envoy doesn't support 0 weight, so instead of passing it to Envoy we just skip such cluster.
 			continue
 		}
-		subsets = append(subsets, envoy_common.ClusterSubset{
+
+		subset := envoy_common.ClusterSubset{
 			ClusterName: service,
 			Weight:      destination.Weight,
 			Tags:        destination.Destination,
-		})
+		}
+		for _, ep := range proxy.OutboundTargets[service] {
```

Comment:
Is there any situation where we can mix external service endpoints with non external here? If not, can we check only first endpoint so we don't have to iterate over the whole collection?